src/specializations/Line.jl

```diff
@@ -24,32 +27,38 @@ function integral(
     t′(x) = (1 + x^2) / (1 - x^2)^2
 
     # Integrate f along the Line
-    domainunits = _units(line(0))
-    integrand(x) = f(line(t(x))) * t′(x) * domainunits
+    differential(line, x) = t′(x) * _units(line(0))
+    integrand(x) = f(line(t(x))) * differential(line, x)
     return sum(w .* integrand(x) for (w, x) in zip(ws, xs))
 end
 
 function integral(
         f::F,
         line::Meshes.Line,
         rule::GaussKronrod;
+        diff_method::DM = Analytical(),
         FP::Type{T} = Float64
-) where {F <: Function, T <: AbstractFloat}
+) where {F <: Function, DM <: DifferentiationMethod, T <: AbstractFloat}
+    _guarantee_analytical(Meshes.Line, diff_method)
+
     # Normalize the Line s.t. line(t) is distance t from origin point
     line = Meshes.Line(line.a, line.a + Meshes.unormalize(line.b - line.a))
 
     # Integrate f along the Line
     domainunits = _units(line(0))
-    integrand(t) = f(line(t)) * domainunits
+    integrand(t) = f(line(t)) * domainunits       # TODO differential
```

Comment:
There is an unresolved TODO here.

Reply:
I just made a commit removing this comment; just waiting on the current Actions to complete. I previously made an effort to tweak all the other special-case methods to make it obvious which part of the integrand expression was essentially just a baked-in `differential(geometry, ts, ::Analytical)`. The `GaussKronrod` methods, however, turn out to be special cases because no domain transform was required since `quadgk` can integrate on $(-\infty, \infty)$ directly.